src/urllib3/request.py

```diff
@@ -84,6 +86,18 @@ def request(
 
         urlopen_kw["request_url"] = url
 
+        if json is not None and body is not None:
+            raise TypeError(
+                "request got values for both 'json' and 'body', can only specify one."
+            )
+
+        if json is not None:
+            if headers is None:
+                headers = self.headers
+                headers["Content-Type"] = "application/json"
+
+            body = json_serializer.dumps(json, separators=(",", ":")).encode("utf-8")
```

Comment:
May I suggest using `ensure_ascii=False` here? IMO there is no good reason to escape to ASCII except for uglifying the output.